src/main/java/randoop/generation/ObjectPool.java

```diff
@@ -0,0 +1,206 @@
+package randoop.generation;
+
+import static randoop.util.EquivalenceChecker.equivalentTypes;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import randoop.DummyVisitor;
+import randoop.ExecutionOutcome;
+import randoop.NormalExecution;
+import randoop.sequence.ExecutableSequence;
+import randoop.sequence.Sequence;
+import randoop.test.DummyCheckGenerator;
+import randoop.types.Type;
+import randoop.util.ListOfLists;
+import randoop.util.SimpleArrayList;
+import randoop.util.SimpleList;
+
+/**
+ * A class representing a pool of objects, each associated with a list of sequences. Used in
+ * randoop.generation.Detective.
```

Comment:
The comment here has been reworked. I decided to use a list rather than a single element because there can be occasions where a method requires multiple inputs of the same type (e.g. foo(T t1, T t2)) and there are more than one sequence that generates the same object of type T in the object pool. Having a list here instead of a single element may contribute to the diversity of statements used in generating the resulting sequence especially when there is only one object of type T in the main and secondary object pools as the same object will be chosen but we have different sequences for it, potentially improving generated test's coverage. Other than this there are other reasons like performance that favors using a single element. I think I need to think a bit more about this to be more certain and hope I can find a time to discuss it with you.